synapseclient/core/credentials/credential_provider.py

```diff
@@ -104,47 +103,66 @@ def _create_synapse_credential(
 
 
 class UserArgsCredentialsProvider(SynapseCredentialsProvider):
-    """
-    Retrieves auth info from user_login_args during a CLI session.
-    """
+    """Retrieves authentication information from user_login_args during a CLI session."""
 
-    def _get_auth_info(
-        self, syn: "Synapse", user_login_args: Dict[str, str]
-    ) -> Tuple[str, str]:
-        return (
-            user_login_args.username,
-            user_login_args.auth_token,
-        )
+    def _get_auth_info(self, syn: "Synapse", user_login_args: UserLoginArgs) -> Tuple[Optional[str], Optional[str]]:
+        """
+        Retrieves authentication information from user_login_args during a CLI session.
 
+        Args:
+            syn (Synapse): Synapse client instance.
+            user_login_args (UserLoginArgs): Object containing login credentials.
+
+        Returns:
+            Tuple[Optional[str], Optional[str]]: A tuple containing username and token,
+            where either or both values may be None.
+        """
+
+        token: Optional[str] = user_login_args.auth_token
+        username: Optional[str] = user_login_args.username if user_login_args.username else None
```

Comment:
This ternary statement shouldn't be needed. We should be able to just do `username: Optional[str] = user_login_args.username`